src/10_Introduction/AMP_for_E-Commerce_Getting_Started.html

```diff
@@ -234,5 +234,30 @@
 
   -->
 
+  <!--
+    ## Client-side Filtering and Sorting
+
+    It is possible to implement client-side filtering and sorting by using `amp-list`, `amp-state` and `amp-bind`. The `amp-state` is initially
+    setup with data from an endpoint then everytime the user select a choice the variable `filteredProducts` is set
+    to the result of the filter expression. `filteredProducts` is then used as `src` of `amp-list`.
+    The alternative to this approach is using server-side filtering and sorting which we explain in the [product sample](https://ampbyexample.com/samples_templates/product_browse_page/)
+  -->
+  <div class="m1">
+    <amp-state id="allProducts" src="/json/related_products.json"></amp-state>
+    <h3>Select your favourite color</h3>
+    <select class="mt1 ampstart-input" on="change:AMP.setState({filteredProducts: allProducts.items.filter(a => event.value == 'all' ? true : a.color == event.value)})">
+        <option value="all" selected="">All</option>
+        <option value="red">red</option>
+        <option value="green">green</option>
+        <option value="yellow">yellow</option>
+        <option value="orange">orange</option>
+    </select>
+    <amp-list credentials="include" width="auto" height="100" layout="fixed-height" src="/json/related_products.json" [src]="filteredProducts" >
```

Comment:
Can you make the initial height match the number of items + update the height based on the filter results.

Reply:
done